core/src/main/scala/org/apache/pekko/persistence/jdbc/journal/dao/JournalQueries.scala

```diff
@@ -50,49 +54,33 @@ class JournalQueries(
     }
   }
 
-  private def selectAllJournalForPersistenceIdDesc(persistenceId: Rep[String]) =
-    selectAllJournalForPersistenceId(persistenceId).sortBy(_.sequenceNumber.desc)
-
   private def selectAllJournalForPersistenceId(persistenceId: Rep[String]) =
     JournalTable.filter(_.persistenceId === persistenceId).sortBy(_.sequenceNumber.desc)
 
   def delete(persistenceId: String, toSequenceNr: Long) = {
     JournalTable.filter(_.persistenceId === persistenceId).filter(_.sequenceNumber <= toSequenceNr).delete
   }
 
-  def markJournalMessagesAsDeleted(persistenceId: String, maxSequenceNr: Long) =
+  def markSeqNrJournalMessagesAsDeleted(persistenceId: String, sequenceNr: Long) =
     JournalTable
       .filter(_.persistenceId === persistenceId)
-      .filter(_.sequenceNumber <= maxSequenceNr)
+      .filter(_.sequenceNumber === sequenceNr)
       .filter(_.deleted === false)
       .map(_.deleted)
       .update(true)
 
   private def _highestSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
     selectAllJournalForPersistenceId(persistenceId).take(1).map(_.sequenceNumber).max
 
-  private def _highestMarkedSequenceNrForPersistenceId(persistenceId: Rep[String]): Rep[Option[Long]] =
-    selectAllJournalForPersistenceId(persistenceId).filter(_.deleted === true).take(1).map(_.sequenceNumber).max
-
-  val highestSequenceNrForPersistenceId = Compiled(_highestSequenceNrForPersistenceId _)
-
-  val highestMarkedSequenceNrForPersistenceId = Compiled(_highestMarkedSequenceNrForPersistenceId _)
-
-  private def _selectByPersistenceIdAndMaxSequenceNumber(persistenceId: Rep[String], maxSequenceNr: Rep[Long]) =
-    selectAllJournalForPersistenceIdDesc(persistenceId).filter(_.sequenceNumber <= maxSequenceNr)
-
-  val selectByPersistenceIdAndMaxSequenceNumber = Compiled(_selectByPersistenceIdAndMaxSequenceNumber _)
-
-  private def _allPersistenceIdsDistinct: Query[Rep[String], String, Seq] =
-    JournalTable.map(_.persistenceId).distinct
-
-  val allPersistenceIdsDistinct = Compiled(_allPersistenceIdsDistinct)
-
-  def journalRowByPersistenceIds(persistenceIds: Iterable[String]): Query[Rep[String], String, Seq] =
-    for {
-      query <- JournalTable.map(_.persistenceId)
-      if query.inSetBind(persistenceIds)
-    } yield query
+  private def _highestSequenceNrForPersistenceIdBefore(
+      persistenceId: Rep[String],
+      maxSequenceNr: Rep[Long]): Rep[Long] =
+    selectAllJournalForPersistenceId(persistenceId)
+      .filter(_.sequenceNumber <= maxSequenceNr)
+      .take(1)
+      .map(_.sequenceNumber)
+      .max
```

Comment:
Seem like we could avoid `max` function here, I have compared `max` and `top 1` execute plans, and the former seems to cost more.

## TOP 1

```
Limit  (cost=0.29..0.35 rows=1 width=8)
  ->  Index Only Scan Backward using event_journal_pkey on event_journal  (cost=0.29..12.29 rows=200 width=8)
        Index Cond: ((persistence_id = 'my-7'::text) AND (sequence_number <= 10000))

```
## Max 

```
Result  (cost=0.35..0.36 rows=1 width=8)
  InitPlan 1 (returns $0)
    ->  Limit  (cost=0.29..0.35 rows=1 width=8)
          ->  Index Only Scan Backward using event_journal_pkey on event_journal  (cost=0.29..12.79 rows=200 width=8)
                Index Cond: ((persistence_id = 'my-7'::text) AND (sequence_number IS NOT NULL) AND (sequence_number <= 10000))

```

Reply:
**The SQL before update.**

```
select max(x2.x3) from (select "sequence_number" as x3 from "event_journal" where ("persistence_id" = ?) and ("sequence_number" <= ?) order by "sequence_number" desc limit 1) x2
```
**The SQL after update.**

```
select "sequence_number" from "event_journal" where ("persistence_id" = ?) and ("sequence_number" <= ?) order by "sequence_number" desc limit 1

```